tests/deepsparse/pipelines/test_dynamic_batch_pipeline.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import concurrent.futures
+import itertools
+from concurrent.futures import ThreadPoolExecutor
+
+import numpy
+
+import pytest
+from data_helpers import create_test_inputs
+from deepsparse import Pipeline
+
+
+@pytest.fixture()
+def executor():
+    """
+    An Auto-delete fixture for yielding a concurrent.futures.ThreadPoolExecutor()
+    object, uses cpu_count + 4 threads, limited 32 to avoid consuming surprisingly
+    large resource on many core machines
+    """
+    yield ThreadPoolExecutor()
+
+
+supported_tasks = [
+    "text_classification",
+    "token_classification",
+    "yolo",
+    "image_classification",
+]
+
+batch_sizes = [
+    1,
+    2,
+    10,
+]
+
+tasks_batch_size_pairs = list(itertools.product(supported_tasks, batch_sizes))
```

Comment:
nit: globals should be capitalized, private globals also should have preceding `_`